### PR TITLE
8343336: Add persistentScrollBars preference

### PR DESCRIPTION
**Pull request by mstr2**
This PR adds another accessibility preference that is available on all supported desktop platforms:

1. **Windows**: Settings -> Accessibility -> Visual Effects -> Always show scroll bars
2. **macOS**: System Settings -> Appearance -> Show scroll bars
3. **Ubuntu**: Settings -> Accessibility -> Seeing -> Always show scroll bars

When the OS setting is changed, all platforms report updates that can be observed in JavaFX.
The manual test application `test/manual/events/PlatformPreferencesChangedTest` can be used to test the feature.

/reviewers 2
/csr

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8343451](https://bugs.openjdk.org/browse/JDK-8343451) to be approved
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8343336](https://bugs.openjdk.org/browse/JDK-8343336): Add persistentScrollBars preference (**Enhancement** - P4)
 * [JDK-8343451](https://bugs.openjdk.org/browse/JDK-8343451): Add persistentScrollBars preference (**CSR**)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1618/head:pull/1618` \
`$ git checkout pull/1618`

Update a local copy of the PR: \
`$ git checkout pull/1618` \
`$ git pull https://git.openjdk.org/jfx.git pull/1618/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1618`

View PR using the GUI difftool: \
`$ git pr show -t 1618`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1618.diff">https://git.openjdk.org/jfx/pull/1618.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1618#issuecomment-2448969624)
</details>

### CODE REVIEW

**Comment by andy-goryachev-oracle** [comment]
This looks like a useful data point.

I am curious as to what are the criteria for including or excluding a particular preference?  What is our long term goal - should we support all the preferences available on a particular platform, or the intersection, or what?  May be it's something to discuss first on the mailing list.

**Comment by mstr2** [comment]
My thinking is that a particular preference should be useful and almost universally supported on the desktop platforms to make it into JavaFX API. The number of preferences we might consider are pretty limited, so I don't expect `Platform.Preferences` to explode in size and scope.

I think this is a minor addition that doesn't change anything substantial about `Platform.Preferences`, and as such doesn't need a major discussion since the arguments are basically the same since the introduction of platform preferences. However, if you or others are interested in having a discussion, we can of course do that.

**Comment by mstr2** [comment]
Changed the naming from `reducedScrollBars` to `persistentScrollBars`, as this matches the logic of the OS settings more closely.

**Comment by kevinrushforth** [comment]
> My thinking is that a particular preference should be useful and almost universally supported on the desktop platforms to make it into JavaFX API.

This seems like the right criteria to me.

> The number of preferences we might consider are pretty limited, so I don't expect `Platform.Preferences` to explode in size and scope.

Good.

> I think this is a minor addition that doesn't change anything substantial about `Platform.Preferences`, and as such doesn't need a major discussion since the arguments are basically the same since the introduction of platform preferences. However, if you or others are interested in having a discussion, we can of course do that.

With this explanation, I agree.

**Comment by kevinrushforth** [comment]
Reviewers: @andy-goryachev-oracle @lukostyra 

(I'll look over the public API and the CSR)

File: modules/javafx.graphics/src/main/java/javafx/application/Platform.java [hunk 1]
```diff
@@ -586,6 +586,20 @@ public static Preferences getPreferences() {
...
+         * Specifies whether applications should always show scroll bars. If not set, an application may
+         * choose to hide scroll bars that are not actively used, or make them smaller or less noticeable.
+         * <p>
+         * If the platform does not report this preference, this property defaults to {@code false}.
+         *
+         * @return the {@code persistentScrollBars} property
+         * @defaultValue {@code false}
+         * @since 24
+         */
+        ReadOnlyBooleanProperty persistentScrollBarsProperty();
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/java/javafx/application/Platform.java:599]: Do you also plan to add new platform-specific preferences or do you think this is unnecessary?

Comment (mstr2) [reply modules/javafx.graphics/src/main/java/javafx/application/Platform.java:602]: I added the respective platform-specific keys to the `Preferences` documentation.

**Review by andy-goryachev-oracle — COMMENTED** [review]
the code changes look good, and both the property and the key reflect the platform preference on macOS M1 14.7.

I'll test windows tomorrow, and if someone can test it on linux, that would be fantastic!

**Review by andy-goryachev-oracle — APPROVED** [review]
looks good on win11, both property and windows key get updated in the monkey tester.

**Review by kevinrushforth — APPROVED** [review]
Looks good. In addition to reviewing the code and the docs, I tested it on all platforms. On Linux I tested it on Ubuntu 24.04  and on a _very_ old 16.04. There is a minor bug on 16.04 (a warning printed), but it could be done as a follow-up if you prefer.

I reviewed the CSR, so it is ready to finalize.

File: modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp [hunk 2]
```diff
@@ -121,6 +121,10 @@ jobject PlatformSupport::collectPreferences() const {
         gboolean enableAnimations = true;
         g_object_get(settings, "gtk-enable-animations", &enableAnimations, NULL);
         putBoolean(env, prefs, "GTK.enable_animations", enableAnimations);
+
+        gboolean overlayScrolling = true;
+        g_object_get(settings, "gtk-overlay-scrolling", &overlayScrolling, NULL);
```
[the remark this answers is not in the document]
Comment (mstr2) [reply modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp]: I've added a runtime check for the `gtk-overlay-scrolling` property.

File: modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp [hunk 3]
```diff
@@ -121,6 +121,12 @@ jobject PlatformSupport::collectPreferences() const {
         gboolean enableAnimations = true;
         g_object_get(settings, "gtk-enable-animations", &enableAnimations, NULL);
         putBoolean(env, prefs, "GTK.enable_animations", enableAnimations);
+
+        if (g_object_class_find_property(G_OBJECT_GET_CLASS(settings), "gtk-overlay-scrolling")) {
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp:125]: Looks good, presuming that we want the property to be absent on systems that don't support this setting.

**Review by kevinrushforth — APPROVED** [review]
The warning is gone on my Ubuntu 16.04 system. I note that the `GTK.overlay_scrolling` preference is not reported on that system (which seems fine) and that the `persistentScrollBarsProperty` is always `false`. Everything still works as expected on systems with a newer GTK.

File: modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp (hunk 3 above)
Comment (mstr2) [reply modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp:125]: Yes, a property that isn't available shouldn't be included in the preferences map. We've documented that in `Platform.Preferences`:
```java
     * The preferences that are reported by the platform may be dependent on the operating system version
     * and its current configuration, so applications should not assume that a particular preference is
     * always available.
```

Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/native-glass/gtk/PlatformSupport.cpp:125]: Yes, so this is fine.

**Review by andy-goryachev-oracle — APPROVED** [review]
ship it!

**Pull request closed without merging**